Add show family feature

scara82 opened the pull request:
**Please consider only the commits from the 4th of February (the last 8 commits) because this branch comes from the ms-ui-interactions not merged yet** 

This is a small PR that:
- Adds feature specs for the admin view of the specific family page
- Adds families controller spec for show method
- Adds show route
- Adds show.html.erb
- Adds families#show.

Thank you!

Commit:
Add spec files for controller testing

Commit:
Add simple form gem and flash messages to rend it

Commit:
Add families routes for new, create, index and root path with corresponding methods in the controller

Commit:
Add in Domus module repsonders_controller and action_controller config

Commit:
Add specs for families#create

Commit:
Add features spec folder and features/families_spec.rb file

Commit:
Rubocop fixes

Commit:
Add spec for families/show

Commit:
Change test for families#show

Commit:
Add route for show and families#show method

Commit:
Add families/show.html.erb

Commit:
Add features specs for Admin that wants to view one family page

Commit:
Modify the test, visit the correct page and add test for presence of the title and family last_name

Commit:
Add title with translation and render the family last name for testing purpose in families/show.html.erb

apalski commented on spec/features/admin_views_one_family_spec.rb:
```diff
@@ -0,0 +1,12 @@
+    expect(page).to have_text(I18n.t("families.show.title"))
+    expect(page).to have_text("Yellow")
+  end
+end
```
Looks good!

apalski commented on spec/controllers/families_controller_spec.rb:
```diff
@@ -0,0 +1,38 @@
+      expect(response).to render_template("show")
+    end
+  end
+end
```
looks good!

apalski commented on app/controllers/families_controller.rb:
```diff
@@ -0,0 +1,34 @@
+  def show
+    @family = Family.find(params[:id])
+  end
+
```
Good!

apalski commented on app/views/families/show.html.erb:
```diff
@@ -0,0 +1,2 @@
+<h1><%= I18n.t("families.show.title") %></h1>
+<%= @family.last_name %>
```
Good!

apalski commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
+  resources :families, only: %i[index show new create]
+
+  root "families#index"
 end
```
Good!

elle commented on app/views/families/index.html.erb:
```diff
@@ -0,0 +1,3 @@
+<a href="families/new"><%= t ".link" %></a>
+<p><%= t ".title" %></p>
```
This is a style comment: I usually prefer to use parens, but up to you

Or at least be consistent across the project. The :new view uses parens

scara82 replied:
👌

elle commented on app/views/families/new.html.erb:
```diff
@@ -0,0 +1,13 @@
+  <%= f.input :animals %>
+  <%= f.input :capability %>
+  <%= f.input :prof, as: :radio_buttons %>
+  <%= f.input :preferred_gender, as: :radio_buttons, collection: ["Female", "Male"] %>
```
Use translations?

scara82 replied:
Done! :)

elle commented on config/environments/production.rb:
```diff
@@ -1,4 +1,7 @@
 Rails.application.configure do
+  config.middleware.use ::Rack::Auth::Basic do |u, p|
+    [u, p] == [ENV["BASIC_AUTH_USERNAME"], ENV["BASIC_AUTH_PASSWORD"]]
```
👍

elle commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  root :to => '/'
+  resources :families, only: %i[index show new create]
```
I usually just prefer to use: `only: [:index, :show, :new, :create]`

You could disable this rule for Rubocop...

scara82 replied:
👌

elle commented on spec/features/admin_adds_a_family_spec.rb:
```diff
@@ -0,0 +1,16 @@
+
+    fill_in "family_last_name", with: "Scaramella"
+    fill_in "family_capability", with: 2
+    click_on I18n.t("families.new.submit")
```
👍

elle commented on spec/features/admin_adds_a_family_spec.rb:
```diff
@@ -0,0 +1,16 @@
+    expect(page).to have_text(
+      I18n.t("flash.actions.create.notice", resource_name: "Family")
+    )
+    expect(current_path).to eq families_path
```
Maybe also add a check to see that "Scaramella" is on the page as well??

scara82 replied:
👌

elle commented on spec/features/admin_views_one_family_spec.rb:
```diff
@@ -0,0 +1,12 @@
+    visit "/families/#{family.id}"
+
+    expect(page).to have_text(I18n.t("families.show.title"))
+    expect(page).to have_text("Yellow")
```
👍

elle commented on spec/features/family_spec.rb:
```diff
@@ -0,0 +1 @@
+require "rails_helper"
```
Is this file not used anymore? in that case, remove it?

scara82 replied:
Oh, yes, you're right. I will delete it. Thanks

elle commented on spec/rails_helper.rb:
```diff
@@ -1,4 +1,5 @@
 require "spec_helper"
+# require "capybara/rspec"
```
Remove comment?

scara82 replied:
👌

elle commented on spec/spec_helper.rb:
```diff
@@ -1,3 +1,5 @@
+require "capybara/rspec"
```
This does not belong here. Not sure if you need it. But if you do, a suggestion to move to `spec/support/capybara_webkit.rb`?

scara82 replied:
Actually I don't, spec pass also without. Deleted, thanks!

Commit:
Change rubocop test to avoid Style/SymbolArray: Use %i or %I for an array of symbols.

Commit:
Change the array for the resources in symbols after changing rubocop preferences against that

Commit:
Add translation for the families.new page to use f for Female and M for Male

Commit:
Add test into feature spec to check if the page shows the last_name of the family just created

Commit:
Delete unecessary require capybara/rspec

Commit:
Delete unecessary file for generic features/family_spec

Commit:
Clean form styling

- Using simple_form .input class for fields styling
- Thus removing the need to call html_wrapper

Commit:
Clean up translation keys

- TODO
  - Update the views to use the new keys structure

Commit:
Add default rake task to run rspec and rubocop

elle commented on .rubocop.yml:
```diff
@@ -15,7 +15,7 @@ Layout/IndentationConsistency:
 Metrics/MethodLength:
   Exclude:
     - "app/controllers/families_controller.rb"
```
This line should be removed. Maybe instead, increase the max lines to about 11?

scara82 replied:
👌

elle commented on app/assets/javascripts/families.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
I thought we removed this file. Should be removed

scara82 replied:
👌

elle commented on app/assets/stylesheets/_flash.css:
```diff
@@ -0,0 +1,6 @@
+   padding: 5px;
+   border: 1px solid blue;
+   width: fit-content;
+ }
```
I think this file was moved. So this file can be removed

scara82 replied:
👌

elle commented on app/views/families/show.html.erb:
```diff
@@ -0,0 +1,2 @@
+<h1><%= I18n.t("families.show.title") %></h1>
```
I believe you don't need to call the `I18n` here. And also just use `t(".title")`

scara82 replied:
👌

elle commented on config/routes.rb:
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
-  resources :families, only: %i[index new create]
+  resources :families, only: [:index, :show, :new, :create]
```
👍

Commit:
Change translation grouping and applyes it

Commit:
Delete empty line

elle commented on Rakefile:
```diff
@@ -1,3 +1,8 @@
+
+task :default do
+  system("rspec")
+  system("rubocop -S")
```
👍

elle commented on app/views/application/_header.html.erb:
```diff
@@ -1,16 +1,16 @@
   <ul class="main-nav">
     <li>
-      <%= link_to t("families.index.home"), root_path %>
+      <%= link_to t("families.header.links.home"), root_path %>
```
I think the key should be: "header.links.home", rather than being nested instead the `families` key

scara82 replied:
👌

elle commented on config/locales/en.yml:
```diff
@@ -1,34 +1,31 @@
+    new:
+      title: "Add a family"
       submit: "Add"
+    header:
```
Yeah, pull it out one more level :)

scara82 replied:
👌

scara82 replied:
👍

elle commented on spec/controllers/families_controller_spec.rb:
```diff
@@ -25,4 +25,14 @@
   end
+
+  describe "GET /families/:id" do
+    it "returns the show family" do
```
Since there is a feature spec for it, I would remove this spec.

scara82 replied:
👌

elle commented on spec/features/admin_views_one_family_spec.rb:
```diff
@@ -0,0 +1,12 @@
+  scenario "successfully" do
+    family = create(:family, last_name: "Yellow")
+
+    visit "/families/#{family.id}"
```
Could you use `family_path(family)` here?

scara82 replied:
👍

Commit:
Delete families.coffee

Commit:
Move the header traslations outside the families and update the path into _header.html.erb

Commit:
Add translation for show.title

Commit:
Replace the translation for new.title

Commit:
Remove the useless test for GET families/:id because already into features spec

Commit:
Replace the UR with the family_path

Commit:
Remove stylesheet/_flash.css that has been renamed in _flashes.scss and moved in components folder

Commit:
Change Rubocop implementation for MethodLenght

scara82 commented:
Thanks @elle and @apalski for your feedbacks! I'm merging now! 😏
